### DISCUSSION TITLE
Running serverpod with https

### DISCUSSION THREAD

**Discussion by hoeverma** [serverpod/serverpod#3087] [Q&A] [answered]
Hi!
I currently try to configure my serverpod application so that I can use it via https, instead of http. I changed development.yaml port and scheme. But how do I configure the SSL certificates? I use this currently in development...
Any help would be highly appreciated, as I will soon have to go through a test, where a fundamental architecture decision for our coming system will be taken. 

Thank you very much in advance,
Markus

**Comment by Isakdl** [comment]
Locally I would never bother with TSL as it isn't necessary and adds a lot of complexity.
I normally set this up in a production environment at the loadbalancer level, e.g., outside of Serverpod.

**Reply by hoeverma** [reply]
Yes, I did this now also, and it seems this is the most reasonable way to handle it. Please excuse, I will nevertheless mark macMikey's answer as the answer to my question, because he proposed a solution for the problem. Yours is a workaround, but, as I said: I go with the workaround...

**Comment by macMikey** [comment, marked as answer]
or...be lazy and use [caddy](https://caddyserver.com/) as your reverse proxy in front of serverpod.
it's oss, written in go, trivial to set up. it handles all the certs for you, and comms between serverpod and caddy will just continue in boring old http, while comms from caddy to the rest of the universe will be https. You can run it on a raspberry pi, if you want to put it on a separate + cheap piece of gear.
i've got a caddy setup that's been running for around eighteen months with zero issues, and no intervention on my part.

**Reply by hoeverma** [reply]
Thank, you, I tried that also, and yes, it works! But it seems a bit complicated to me, although you are right: It is very easy to setup, and seems to work reliably. I will just go on with http in development...

**Comment by macMikey** [comment]
i would encourage you to continue working through setting the certs and recerts up, yourself. when you're done, you can decide which way is better for you.
